### PR TITLE
Adding MSVC compatibility

### PR DESCRIPTION
**Pull request by MarcoLizza**
I made some minor changes to the source files to add a MSVC (2013) build files and to resolve the platform errors and warnings.

Hope it helps!

### CODE REVIEW

File: src/wren_compiler.c [hunk 1]
```diff
@@ -894,7 +898,7 @@ static void ignoreNewlines(Compiler* compiler)
 // discards any duplicate newlines following it.
 static bool consumeLine(Compiler* compiler, const char* errorMessage)
 {
-  bool result = consume(compiler, TOKEN_LINE, errorMessage);
+  bool result = consume(compiler, TOKEN_LINE, errorMessage) != NULL;
```
Comment (zeckalpha) [inline src/wren_compiler.c]: Would this make more sense inside consume?

File: src/wren_vm.c [hunk 2]
```diff
@@ -959,7 +963,7 @@ static bool runInterpreter(WrenVM* vm)
       // Capture upvalues.
       for (int i = 0; i < prototype->numUpvalues; i++)
       {
-        bool isLocal = READ_BYTE();
+        bool isLocal = READ_BYTE() != 0;
```
Comment (zeckalpha) [inline src/wren_vm.c]: Same here?

File: src/wren_compiler.c (hunk 1 above)
Comment (MarcoLizza) [inline src/wren_compiler.c]: No, it wouldn't. The "consume(...)" function returns a pointer and this is only a special case check. (same applies for the other similiar change).

**Comment by munificent** [comment]
Thank you for doing this! I like that you split it out into multiple commits (woo!) but as a PR it's a bit monolithic to pull in just yet. I :+1: the commits that I think are pretty much ready as is and pushed a few changes that should address some things you had to work around.

**Comment by zeckalpha** [comment]
Sorry, I didn't bother to look at the methods; just noticed the types
didn't change. Looks great!

**Comment by MarcoLizza** [comment]
I'm closing this pull-request. The new one supersedes it.

**Pull request closed without merging**